internal/metrics/chat_completion_metrics.go

```diff
@@ -12,18 +12,14 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/metric"
 
-	"github.com/envoyproxy/ai-gateway/filterapi"
 	"github.com/envoyproxy/ai-gateway/filterapi/x"
 )
 
 // chatCompletion is the implementation for the chat completion AI Gateway metrics.
 type chatCompletion struct {
-	metrics        *genAI
+	*baseMetrics
```

Comment:
i don't think this needs to be a pointer (less allocation:)
```suggestion
	baseMetrics
```